### DISCUSSION TITLE
Using fast component inside a Web Component

### DISCUSSION THREAD

**Discussion by TimO2221** [microsoft/fast#5294] [Q&A] [answered]
Hello, I am trying to use a fast component inside a custom web component but get the following error:
`Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.`

My Web Component looks like this:
```

import { 
    allComponents, 
    provideFASTDesignSystem 
} from "../node_modules/@microsoft/fast-components";

class newelement extends HTMLElement {
    constructor() {
        // Always call super first in constructor
        super();

        provideFASTDesignSystem().register(allComponents);

        this.attachShadow({ mode: 'open' }); // sets and returns 'this.shadowRoot'

        const template = document.createElement('template');
        template.content = `
                <fast-menu id="mainMenu" slot="header-slot" style="display: flex;">
                    <fast-menu-item role="menuitem">People</fast-menu-item>
                    <fast-menu-item disabled role="menuitem">Places</fast-menu-item>
                    <fast-menu-item disabled role="menuitem">Profile</fast-menu-item>
                </fast-menu>
        `;

        this.shadowRoot.append(template.content.cloneNode(true));
  }
}

customElements.define('new-element ', newelement );
```

**Comment by EisenbergEffect** [comment, marked as answer]
I think the issue you have is the module path for the `import` statement. It seems like you are maybe trying to do this directly from a browser script without a bundler. If that's the case, try this:

```js
import { 
    allComponents, 
    provideFASTDesignSystem 
} from "../node_modules/@microsoft/fast-components/dist/esm/index.js";
```

**Reply by TimO2221** [reply]
Yes Eisenberg, I am not using a bundler.

**Reply by EisenbergEffect** [reply]
Let me know if the above fixes the issue for you. And apologies for taking so long to get a response to you.

**Reply by TimO2221** [reply]
I am sorry, I have move away from using this library. I was having to many issues and decided to build my own. I don't know if your solution would have worked or not.
